lib/Lower/Bridge.cpp

```diff
@@ -574,51 +385,338 @@ class FirConverter : public Fortran::lower::AbstractConverter {
+    tripVariable = builder->create<mlir::SubIOp>(location, tripVariable, one);
+    builder->create<fir::StoreOp>(location, tripVariable, info.tripVariable);
+    genFIRUnconditionalBranch(info.headerBlock);
+  }
+
+  /// Generate structured or unstructured FIR for an IF construct.
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::IfConstruct &) {
+    if (eval.LowerAsStructured()) {
+      // Structured fir.where nest.
+      fir::WhereOp where;
+      mlir::OpBuilder::InsertPoint insertionPoint{
+          builder->saveInsertionPoint()};
+      for (auto &e : *eval.evaluationList) {
+        if (auto *s{e.getIf<Fortran::parser::IfThenStmt>()}) {
+          // fir.where op
+          genWhereCondition(where, s);
+        } else if (auto *s{e.getIf<Fortran::parser::ElseIfStmt>()}) {
+          // otherwise block, then nested fir.where
+          switchInsertionPointToOtherwise(where);
+          genWhereCondition(where, s);
+        } else if (e.isA<Fortran::parser::ElseStmt>()) {
+          // otherwise block
+          switchInsertionPointToOtherwise(where);
+        } else {
+          genFIR(e, false);
+        }
+      }
+      builder->restoreInsertionPoint(insertionPoint);
+      return;
+    }
+
+    // Unstructured branch sequence.
+    for (auto &e : *eval.evaluationList) {
+      if (auto *s{e.getIf<Fortran::parser::IfThenStmt>()}) {
+        maybeStartBlock(e.block);
+        genFIRConditionalBranch(
+            std::get<Fortran::parser::ScalarLogicalExpr>(s->t),
+            e.lexicalSuccessor, e.controlSuccessor);
+      } else if (auto *s{e.getIf<Fortran::parser::ElseIfStmt>()}) {
+        startBlock(e.block);
+        genFIRConditionalBranch(
+            std::get<Fortran::parser::ScalarLogicalExpr>(s->t),
+            e.lexicalSuccessor, e.controlSuccessor);
+      } else {
+        genFIR(e);
+      }
+    }
+  }
+
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::CaseConstruct &) {
+    TODO();
+  }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::SelectRankConstruct &) {
+    TODO();
+  }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::SelectTypeConstruct &) {
+    TODO();
+  }
+
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::WhereConstruct &) {
+    TODO();
+  }
 
   /// Lower FORALL construct (See 10.2.4)
-  void genFIR(const Fortran::parser::ForallConstruct &forall) {
-    auto &stmt = std::get<
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::ForallConstruct &forall) {
+    auto &stmt{std::get<
         Fortran::parser::Statement<Fortran::parser::ForallConstructStmt>>(
-        forall.t);
+        forall.t)};
     setCurrentPosition(stmt.source);
-    auto &fas = stmt.statement;
-    auto &ctrl =
+    auto &fas{stmt.statement};
+    auto &ctrl{
```

Comment:
As mentioned via slack, doing this was too much for yet another merge.  (The only way to do this is to start over on the entire file.  Since most bridge files currently use the braces mechanism, it also seems a little odd to single out this file.  Maybe we need to bring up this whole f18 vs. llvm style business in our weekly meeting?  And the do (redo) this separately.

Reply:
Done for the moment; may need to be undone ...

Reply:
Yep.